src/Components/Endpoints/src/Builder/ResourcePreloadCollection.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
+
+internal class ResourcePreloadCollection
+{
+    private readonly Dictionary<string, StringValues> _storage = new();
+
+    public ResourcePreloadCollection(ResourceAssetCollection assets)
+    {
+        if (assets != null)
```

Comment:
Do we need this null check? We are the only ones creating this type and the parameter is marked as non nullable